### PR TITLE
Merge duplicate files in monorepo

### PR DESCRIPTION
**Pull request by Buckram123**
Closes ABS-225
### Checklist

- [x] CI is green.
- [x] Changelog updated.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1 +1,3 @@
-**/artifacts/* binary
\ No newline at end of file
+**/artifacts/* binary
+**/schema/*.json linguist-generated=true
+**/docs/book linguist-generated=true
```
Comment (CyberHoward) [inline .gitattributes:3]: this is also in gitignore so probably not needed here?

File: justfile [hunk 2]
```diff
@@ -49,3 +49,9 @@ schema-modules:
     (cd $path; cargo run --example schema --features schema); 
   done
   set +e
+
+schema: schema-modules
```
Comment (CyberHoward) [inline justfile:53]: We should have one for framework as well

Comment (CyberHoward) [reply justfile:53]: there's a cargo-workspace command that lets you run commands on every member of a workspace.

Comment (CyberHoward) [reply justfile:53]: https://crates.io/crates/cargo-workspaces

Comment (CyberHoward) [reply justfile:53]: See the justfile in modules!

Comment (CyberHoward) [reply justfile:53]:
```
# TODO this doesn't work for contracts like the croncat one which is the croncat prefix
publish-schemas:
  SCHEMA_OUT_DIR=$(cd ../../schemas && echo "$PWD") \
    cargo ws exec --no-bail bash -lc 'cargo schema && { jq -r .contract_version schema/module-schema.json > _version.txt; outdir="$SCHEMA_OUT_DIR/abstract/${PWD##*/}/$(cat _version.txt)"; mkdir -p "$outdir"; rm -rf "schema/raw"; cp -a "schema/." "$outdir"; rm _version.txt; }'

```

Comment (Buckram123) [reply justfile:53]: > ```
> # TODO this doesn't work for contracts like the croncat one which is the croncat prefix
> publish-schemas:
>   SCHEMA_OUT_DIR=$(cd ../../schemas && echo "$PWD") \
>     cargo ws exec --no-bail bash -lc 'cargo schema && { jq -r .contract_version schema/module-schema.json > _version.txt; outdir="$SCHEMA_OUT_DIR/abstract/${PWD##*/}/$(cat _version.txt)"; mkdir -p "$outdir"; rm -rf "schema/raw"; cp -a "schema/." "$outdir"; rm _version.txt; }'
> ```

There's similar script to this one in framework justfile, we can execute that, if we don't plan to remove justfile's

Comment (CyberHoward) [reply justfile:53]: Yeah, but I don't even know what that command does. 
I think we only need to publish schemas for the modules but not sure. 

@adairrr

Comment (adairrr) [reply justfile:53]: It runs cargo schema for each of the modules, then copies the contents of the schemas for those modules to the proper directory (name + version) in the schemas folder (which in writing the command was in the `../../schemas`. It's a bit hacky and we should move it to a github action.

Comment (CyberHoward) [reply justfile:53]: I say we move it to a `.sh` script and call that from the github action. Easier to maintain and we can run it locally as well. 

@Buckram123 Could you do that?

Comment (Buckram123) [reply justfile:53]: Do we want copy schemas to `../../schemas` in the same script?

File: framework/justfile [hunk 3]
```diff
@@ -54,6 +54,7 @@ migrate +CHAINS:
   just run-script migrate {{CHAINS}}
 
 publish-schemas version:
+  mkdir -p ../../schemas
```
Comment (CyberHoward) [inline framework/justfile:57]: We don't want a script to create a dir that's outside of the repo right?

Comment (Buckram123) [reply framework/justfile]: Right

File: scripts/framework-schema.sh
Comment (CyberHoward) [inline scripts/framework-schema.sh:1]: What does this script do?

File: scripts/modules-schema.sh
Comment (CyberHoward) [inline scripts/modules-schema.sh:1]: What does this script do?

File: justfile (hunk 2 above)
Comment (CyberHoward) [reply justfile:43]: I don't know, should all the schemas live in to one dir? What's the desired setup here? 
@adairrr

File: scripts/framework-schema.sh
Comment (Buckram123) [reply scripts/framework-schema.sh:1]: generate schemas for all framework contracts and copy to the `abstract/../schemas` (if exists)

File: scripts/modules-schema.sh
Comment (Buckram123) [reply scripts/modules-schema.sh:1]: generate schemas for all modules and copy to the `abstract/../schemas` (if exists)

File: schema/execute_msg.json
Comment (Buckram123) [inline schema/execute_msg.json:1]: Is this file generated by app-template supposed to be here? @CyberHoward

File: .gitattributes (hunk 1 above)
Comment (Buckram123) [reply .gitattributes]: Yeah, thanks, removed it!

File: .gitignore [hunk 4]
```diff
@@ -1,10 +1,41 @@
...
+**/.DS_Store
+
+# temporary dockerfiles
+**/docker*
+**/.dockerignore
+
+# Schemas
+framework/packages/**/schema
+framework/contracts/**/schema
+./schema
```
Comment (Buckram123) [inline .gitignore:32]:
```suggestion
```
Remove this gitignore line since we decided to move schema inside this repo?

File: schema/execute_msg.json
Comment (CyberHoward) [reply schema/execute_msg.json:1]: good catch!

**Pull request merged**